### PR TITLE
Add a simple way of merging coverage data

### PR DESCRIPTION
**Pull request by abarth**
`flutter test` now has a `--merge-coverage` flag that can be used to merge
coverage data from previous runs, enabling faster iteration cycles.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/test/animation/curves_test.dart [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/animation.dart';
+import 'package:flutter/widgets.dart';
+
+void main() {
+  test('Curve flipped control test', () {
+    Curve ease = Curves.ease;
+    Curve flippedEase = ease.flipped;
+    expect(flippedEase.transform(0.5), lessThan(ease.transform(0.5)));
```
Comment (Hixie) [inline packages/flutter/test/animation/curves_test.dart]: should probably also test that they're equal at 0.0 and 1.0

File: packages/flutter_tools/lib/src/commands/test.dart [hunk 2]
```diff
@@ -83,6 +91,54 @@ class TestCommand extends FlutterCommand {
...
+    String coveragePath = argResults['coverage-path'];
+    File coverageFile = new File(coveragePath)
+      ..createSync(recursive: true)
+      ..writeAsStringSync(coverageData, flush: true);
+    printTrace('wrote coverage data to $coveragePath (size=${coverageData.length})');
+
+    String baseCoverageData = 'coverage/lcov.base.info';
+    if (mergeCoverageData) {
+      if (!os.isLinux) {
+        printError('Merging coverage data is supported only on Linux.');
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/test.dart]: add a comment here saying what's linux-specific about it (and maybe also print that)

File: packages/flutter_tools/lib/src/commands/test.dart [hunk 3]
```diff
@@ -83,6 +91,54 @@ class TestCommand extends FlutterCommand {
...
+        return false;
+      }
+
+      if (!FileSystemEntity.isFileSync(baseCoverageData)) {
+        printError('Missing "$baseCoverageData". Unable to merge coverage data.');
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/test.dart]: suggest running update-packages if you detect that the test is in the flutter repo

Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/test.dart]: maybe file a bug about documenting this for the end-user case?

**Comment by Hixie** [comment]
LGTM

File: packages/flutter_tools/lib/src/commands/test.dart (hunk 3 above)
Comment (abarth) [inline packages/flutter_tools/lib/src/commands/test.dart]: I didn't do either of these things yet.  :)

**Pull request merged**